pkg/webhook/patch.go

```diff
@@ -162,14 +166,11 @@ func addVolume(pod *corev1.Pod, volume corev1.Volume) patchOperation {
 	return patchOperation{Op: "add", Path: path, Value: value}
 }
 
-func addVolumeMount(pod *corev1.Pod, mount corev1.VolumeMount) patchOperation {
-	i := 0
-	// Find the driver or executor container in the pod.
-	for ; i < len(pod.Spec.Containers); i++ {
-		if pod.Spec.Containers[i].Name == config.SparkDriverContainerName ||
-			pod.Spec.Containers[i].Name == config.SparkExecutorContainerName {
-			break
-		}
+func addVolumeMount(pod *corev1.Pod, mount corev1.VolumeMount) *patchOperation {
+	i := findContainer(pod)
+	if i < 0 {
+		glog.Warningf("not able to add VolumeMount %s as Spark container was not found in pod %s", mount.Name, pod.Name)
```

Comment:
I think others who might depend on podTemplates may find it useful to have the container name said out in the warning-log, which is "executor" for Spark 2 and "spark-kubernetes-executor" for Spark 3. It may help to debug problems of this kind. Verbosity in case of error is always useful. :-)

Reply:
We don't parse the `sparkVersion` due to the lack of validation on the field and a proper parser. This is why in `findContainer` we match executor container with one of the two possible container names. Because of that, it's no easy way of knowing which container is missing.